components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FxaActionProcessor.kt

```diff
@@ -0,0 +1,340 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.appservices.fxaclient
+
+import android.util.Log
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * Processes actions sent to [FxaClient.queueAction] and sends events to the [FxaEventHandler]
+ *
+ * See [FxaClient.queueAction] for details on why we handle actions like this.
+ *
+ * [FxaActionProcessor] is "inert" on construction and won't process any events.  Call
+ * [runActionProcessorManager] start event processing.  This creates a manager task that monitors
+ * the action handling loop and restarts it if it throws for any reason.
+ *
+ */
+internal class FxaActionProcessor(
+    private val inner: FirefoxAccount,
+    private val eventHandler: FxaEventHandler,
+    private val tryPersistState: () -> Unit,
```

Comment:
This name surprises me - from the implementation the intent is that it never throws an exception, which is the opposite of what I'd expect of a function name starting with `try`.